### PR TITLE
add plugin: save-refresh

### PR DESCRIPTION
**Pull request by chentsulin**
### save-refresh
 > This feature may be supported after bookshelf v0.14.x: https://github.com/bookshelf/bookshelf/issues/1665
 ```js
user.save({ key: value }, { withRefresh: true });
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +88,16 @@ user.type; // Same as user.get('type');
 user.age; // Same as user.get('age');
 ```
 
+### save-refresh
+
+> This feature may be supported after bookshelf v0.14.x: https://github.com/bookshelf/bookshelf/issues/1665
+
+```js
+user.save(null), { withRefresh: true });
```
Comment (LittleWhiteYA) [inline README.md]: 這邊 null 多了個右括號XD

File: src/plugins/save-refresh.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      // Handle both `"key", value` and `{key: value}` -style arguments.
+      if (firstArg == null || typeof firstArg === 'object') {
+        attrs = firstArg || {};
+        options = secondArg || {};
+      } else {
+        attrs = {
+          [firstArg]: secondArg,
+        };
+        options = thirdArg || {};
+      }
```
Comment (LittleWhiteYA) [inline src/plugins/save-refresh.js]: Shell we just get `withRefresh` in options from arguments ?
`attrs` are already handled by native saved method

Comment (chentsulin) [reply src/plugins/save-refresh.js]: But we need to get options from 2 or 3:  

```
user.save({ key: value }, { withRefresh: true });
user.save(key, value, { withRefresh: true });
```

Comment (LittleWhiteYA) [reply src/plugins/save-refresh.js]: :confused: 
OK, that's fine.

**Pull request merged**